operators/src/processing/vector_join/equi_data_join.rs

```diff
@@ -358,7 +358,10 @@ where
                         Err(e) => return stream::once(async { Err(e) }).boxed(),
                     };
 
-                    let data_query = self.right_processor.query(query, ctx);
+                    let data_query = match self.right_processor.query(query, ctx) {
```

Comment:
So the left stream produced a valid tile but the right stream could not be created. I think the whole `vector_query` should fail instead of creating a single error tile.

Reply:
You can either fail in the method or within the stream.

We can't do the first thing, since we create multiple query processors within the stream.
Each of them could fail and we can't create them in advance.

When we are in the stream, we can't suddenly hop out and let `vector_query` crash.